#15119 Repro: User without data access should be able to use dashboard filters

nemanjaglumac opened the pull request:
### Status
PENDING REVIEW

### What does this PR accomplish?
- Reproduces #15119 

### How to test this manually?
- `yarn test-cypress-open --spec frontend/test/metabase/scenarios/dashboard/dashboard.cy.spec.js`
- Replace `it.skip()` with `it.only()` to run the test in isolation
- The test should fail until the related issue is fixed

### Additional notes:
- Once the issue is fixed, please remove the `.skip` part (unskip the test completely)
- Make sure the test is passing and
- Merge it together with the fix

### Screenshots:
Add filter button is disabled for "nodata" user
![image](https://user-images.githubusercontent.com/31325167/111005390-0014fb80-838b-11eb-8ec9-dbb6c62b706b.png)

nemanjaglumac committed:
Add repro for #15119

flamber reviewed (commented):
I'm unsure if no-data user should be able to see the dropdown - it comes with it's own can of worms.
But the input should at least function properly as a text input without jumping through hoops.

nemanjaglumac committed:
Update test

nemanjaglumac commented:
Updated the test, so it doesn't rely on the dropdown menu. Updated screenshots as well.

NOTE: This test will fail IF "nodata" user can get a dropdown, because placeholder texts are different for when there is a dropdown menu and when there isn't! cc @flamber

flamber reviewed (approved):
As you say, it will fail if we start showing dropdowns, but then we'll update the test in that case.
